dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/transport/AbstractClient.java

```diff
@@ -152,6 +152,25 @@ private synchronized void initConnectStatusCheckCommand() {
                 @Override
                 public void run() {
                     try {
+
+                        /**
+                         * If the provider service is detected offline,
+                         * the client should not attempt to connect again.
+                         *
+                         * issue: https://github.com/apache/incubator-dubbo/issues/3158
+                         */
+                        if(isClosed()) {
```

Comment:
do you feel that refactoring by extracting to a method may increase the readability of code 

```
/**
                         * If the provider service is detected offline,
                         * the client should not attempt to connect again.
                         *
                         * issue: https://github.com/apache/incubator-dubbo/issues/3158
  */
if(cancelFutureIfClientIsOffline(reconnectExecutorFuture)) {
     reconnectExecutorFuture.cancel(true);
}

private boolean cancelFutureIfClientIsOffline(ScheduledFuture future) {
       return isClosed()) ?  (future != null && !future.isCancelled()) :false;
}
```

Reply:
@khanimteyaz nice, I will fix that, :-)

Reply:
thanks.